modules/ROOT/pages/addition/agent-installation/self-registered.adoc

```diff
@@ -0,0 +1,169 @@
+
+An agent in standby mode shows up in the list of agents in NOM UI. To enable the agent to continue it's normal execution, agent needs to be approved from NOM UI as shown below:
+
+image::agents.png[width=800]
+
+To approve the agent click on `...` agent action icon and click `Approve`:
+
+image::agents.png[width=800]
+
+An edit view to optionally update agent name or description will show as below:
+
+NOTE: an unauthorized self-registering agent will have default name derived from it's host information. It's recommended to update this name or use the optional configuration described <<agent-meta-data,here>>.
+
+image::agent-new.png[width=800]
+
+Once the agent is approved, by optionally updating it's name and description, agent will receive initial configuration from NOM server which it persists on the instance host in a file named `nom-agent-config.yaml`.
+
+WARNING: Location of where agent saves it's initial configuration must be of persistent type else agent needs to be reregistered.
+
+Following optional configuration can be used to provide a persistent location for agent intial configuration:
+
+[cols="<,<,<,<",options="header"]
+|===
+| Variable
+| Description
+| Example
+| Default
+
+| `CONFIG_AGENT_CONFIG_PATH`
+| Peristent path to a file on Neo4j instance host
+| "file://path/to/"
+| by default agent tries to save initial configuration in path `NEO4J_CONF` if set or `conf` folder under `NEO4J_HOME` if set, else the file is saved under `.nom` folder in user home directory.
+|===
+
+Apart from above configuration, following optional configuration can be provided as agent meta data:
+
+[[agent-meta-data]]
+[cols="<,<,<",options="header"]
+|===
+| Variable
+| Description
+| Example
+
+| `CONFIG_AGENT_NAME`
+| Optional name for agent to easily differentiate among self-registered agents
+| home-db-agent
+
+| `CONFIG_AGENT_DESCRIPTION`
+| Optional description for agent to easily differentiate among self-registered agents
+| An agent to monitor home db
+|===
+
+== Configuration for agent self registration with mutual authentication
+Following are the required environment variables to be configured to enable self-registration of NOM agent with mutual authentication:
+
+[cols="<,<,<",options="header"]
+|===
+| Variable
+| Description
+| Example
+
+| `CONFIG_SERVER_GRPC_ADDRESS`
+| Server GRPC Address
+| server:9090
+
+| `CONFIG_SERVER_HTTP_ADDRESS`
+| Server HTTP address
+| +++https://server:8080+++
+
+| `CONFIG_CLIENT_CERT`
+| PEM encoded Agent certificate for mutual TLS
+| /path/to/a/pem/file`
+
+| `CONFIG_CLIENT_KEY`
+| PEM encoded Agent key for mutual TLS
+| /path/to/a/pem/file`
+|===
+
+IMPORTANT: *In addition to above configuration, NOM server also needs to be configured to trust the agent certificates as described xref:/installation/server.adoc#config_ref[here]*
+
+=== Self signed certificates for agent
+
+[IMPORTANT]
+====
+Self-signed certificates are not recommended to be used in production environments.
```

Comment:
Is there a reason we're not recommending self-signed certs for mTLS in production? I get it with HTTPS certs that you'd want to have a proper CA signed cert, but for self-registration it shouldn't matter as long as the key is kept safe. Unless I'm missing something?

Reply:
You are right about the agent keys being a maintenance overhead, we can still suggest it though